src/services/PostService.ts

```diff
@@ -223,5 +223,33 @@ export class PostService {
     }, 0);
     return result;
   }
+
+  public async similarPosts(params: UuidParam): Promise<PostModel[]> {
+    return this.transactions.readOnly(async (transactionalEntityManager) => {
+      const postRepository = Repositories.post(transactionalEntityManager);
+      const post = await postRepository.getPostById(params.id);
+      if (!post) throw new NotFoundError('Post not found!');
+      const allPosts = await postRepository.getAllPosts();
+      let posts: PostModel[] = []
+      for (const p of allPosts) {
+        if (post.id != p.id) {
+          const model = await getLoadedModel();
```

Comment:
can we have this line outside the `for (const p of allPosts) {` for loop? we only need to get the loaded model once per function call